File: frontend/src/utils/currency.js
/**
 * Currency utility for Zambian Kwacha (ZMK)
 */

export const formatZMK = (amount, options = {}) => {
  const {
    symbol = 'ZMK',
    decimals = 2,
    compact = false,
    label = 'Kwacha'
  } = options;
  if (amount === null || amount === undefined) return `${symbol} 0.00`;
  if (compact && Math.abs(amount) >= 1000000) {
    return `${symbol} ${(amount / 1000000).toFixed(1)}M`;
  }
  if (compact && Math.abs(amount) >= 1000) {
    return `${symbol} ${(amount / 1000).toFixed(1)}K`;
  }
  return `${symbol} ${amount.toFixed(decimals).replace(/\B(?=(\d{3})+(?!\d))/g, ',')}`;
};

export const formatKwacha = (amount, compact = false) => {
  return formatZMK(amount, { symbol: 'ZMK', compact });
};

export const parseZMK = (value) => {
  if (typeof value === 'number') return value;
  // Remove 'ZMK', commas and convert
  const cleaned = String(value)
    .replace(/[ZMK\s,]/g, '')
    .trim();
  return parseFloat(cleaned) || 0;
};

// Zambian salary guide
export const ZAMBIAN_SALARY_GUIDE = {
  entryLevel: { min: 2500, max: 4000, label: 'Entry Level' },
  midLevel: { min: 4500, max: 8000, label: 'Mid Level' },
  seniorLevel: { min: 8500, max: 15000, label: 'Senior Level' },
  executive: { min: 16000, max: 30000, label: 'Executive' }
};

// Zambian expense categories with typical ranges
export const ZAMBIAN_EXPENSE_CATEGORIES = [
  { name: 'Rent/Mortgage', typicalRange: [1500, 6000], type: 'essential' },
  { name: 'Electricity (ZESCO)', typicalRange: [300, 800], type: 'essential' },
  { name: 'Water', typicalRange: [80, 200], type: 'essential' },
  { name: 'Internet', typicalRange: [350, 600], type: 'essential' },
  { name: 'Mobile Data', typicalRange: [100, 300], type: 'essential' },
  { name: 'Groceries', typicalRange: [800, 2500], type: 'essential' },
  { name: 'Transport', typicalRange: [300, 1500], type: 'essential' },
  { name: 'Restaurants', typicalRange: [200, 800], type: 'want' },
  { name: 'Entertainment', typicalRange: [200, 600], type: 'want' },
  { name: 'Shopping', typicalRange: [300, 1500], type: 'want' },
  { name: 'School Fees', typicalRange: [1000, 5000], type: 'essential' },
  { name: 'Medical', typicalRange: [200, 1500], type: 'essential' },
];

File: backend/app/utils/currency.py
"""
Currency utility for handling Zambian Kwacha (ZMK)
"""

class CurrencyFormatter:
    """Format amounts in Zambian Kwacha"""
    
    CURRENCY_SYMBOL = "ZMK"
    CURRENCY_CODE = "ZMK"
    
    @staticmethod
    def format(amount, include_symbol=True):
        """Format amount as ZMK currency string"""
        if include_symbol:
            return f"ZMK {amount:,.2f}"
        return f"{amount:,.2f}"
    
    @staticmethod
    def format_kwacha(amount):
        """Format with Kwacha symbol"""
        return f"Kwacha {amount:,.2f}"
    
    @staticmethod
    def parse(amount_str):
        """Parse ZMK amount string to float"""
        # Remove 'ZMK', 'Kwacha', commas and convert
        cleaned = amount_str.replace('ZMK', '').replace('Kwacha', '').replace(',', '').strip()
        return float(cleaned)

# Zambian-specific category names (optional)
ZAMBIAN_CATEGORIES = {
    'expense': [
        'Rent/Mortgage',
        'Electricity (ZESCO)',
        'Water (LWSC)',
        'Internet',
        'Mobile Data/Airtime',
        'DSTV/GOtv',
        'Groceries (Shoprite/Spar)',
        'Market Shopping',
        'Restaurants',
        'Transport (Fuel/Bus/Taxi)',
        'Entertainment',
        'Shopping',
        'Personal Care',
        'Home Supplies',
        'Medical',
        'School Fees',
        'Wedding/Gifts',
        'Travel',
        'Miscellaneous'
    ],
    'income': [
        'Salary',
        'Business Income',
        'Rental Income',
        'Bonus',
        'Dividends',
        'Tax Refund',
        'Other Income'
    ]
}

# Realistic Zambian salary ranges (monthly, in ZMW)
ZAMBIAN_SALARY_RANGES = {
    'entry_level': (2500, 4000),
    'mid_level': (4500, 8000),
    'senior_level': (8500, 15000),
    'executive': (16000, 30000)
}

# Typical Zambian expense amounts (in ZMW)
ZAMBIAN_EXPENSE_GUIDE = {
    'Rent (Lusaka)': {
        'low': (1500, 2500),
        'medium': (3000, 5000),
        'high': (6000, 10000)
    },
    'Rent (Other towns)': {
        'low': (800, 1500),
        'medium': (1800, 3000),
        'high': (3500, 6000)
    },
    'Groceries (monthly)': {
        'single': (800, 1500),
        'couple': (1500, 2500),
        'family': (2500, 4000)
    },
    'Transport (monthly)': {
        'bus': (300, 600),
        'taxi': (600, 1200),
        'car': (1500, 3000)
    },
    'Utilities (monthly)': {
        'electricity': (300, 800),
        'water': (80, 200),
        'internet': (350, 600)
    }
}